File: labs/solidworks/swex/pmpage/events/closing/Events.Closing.cs
public class DataModel
{
    public string Text { get; set; }
}

private DataModel m_Data;

private PropertyManagerPageEx<MyPMPageHandler, DataModel> m_Page;

public override bool OnConnect()
{
    m_Data = new DataModel();
    m_Page = new PropertyManagerPageEx<MyPMPageHandler, DataModel>(App);

    m_Page.Handler.Closing += OnClosing;

    return true;
}

private void OnClosing(swPropertyManagerPageCloseReasons_e reason, ClosingArg arg)
{
    if (reason == swPropertyManagerPageCloseReasons_e.swPropertyManagerPageClose_Okay)
    {
        if (string.IsNullOrEmpty(m_Data.Text))
        {
            arg.Cancel = true;
            arg.ErrorTitle = "Insert Note Error";
            arg.ErrorMessage = "Please specify the note text";
        }
    }
}

File: labs/solidworks/swex/pmpage/events/closing/Events.Closing.vb
Public Class DataModel
    Public Property Text As String
End Class

Private m_Data As DataModel
Private m_Page As PropertyManagerPageEx(Of MyPMPageHandler, DataModel)

Public Overrides Function OnConnect() As Boolean
    m_Data = New DataModel
    m_Page = New PropertyManagerPageEx(Of MyPMPageHandler, DataModel)(App)

    AddHandler m_Page.Handler.Closing, AddressOf OnClosing
    Return True
End Function

Private Sub OnClosing(ByVal reason As swPropertyManagerPageCloseReasons_e, ByVal arg As ClosingArg)
    If reason = swPropertyManagerPageCloseReasons_e.swPropertyManagerPageClose_Okay Then

        If String.IsNullOrEmpty(m_Data.Text) Then
            arg.Cancel = True
            arg.ErrorTitle = "Insert Note Error"
            arg.ErrorMessage = "Please specify the note text"
        End If
    End If
End Sub